plugins/kotlin-serialization/kotlin-serialization-ide/src/org/jetbrains/kotlinx/serialization/compiler/extensions/SerializationIDEResolveExtension.kt

```diff
@@ -21,6 +21,9 @@ class SerializationIDEResolveExtension : SerializationResolveExtension() {
     override fun getSyntheticNestedClassNames(thisDescriptor: ClassDescriptor): List<Name> =
         getIfEnabledOn(thisDescriptor) { super.getSyntheticNestedClassNames(thisDescriptor) } ?: emptyList()
 
+    override fun maybeGetSyntheticNestedClassNames(thisDescriptor: ClassDescriptor): List<Name>? =
+        getIfEnabledOn(thisDescriptor) { super.maybeGetSyntheticNestedClassNames(thisDescriptor) } ?: emptyList()
```

Comment:
If `maybeGetSyntheticNestedClassNames` returns null, `emptyList()` still is the result because of `?:`. Is this a desirable behavior?

Reply:
Yes, it was a mistake. Thank you for pointing this out.